### PR TITLE
Contribution export (xlsx,csv): add more fields

### PR DESCRIPTION
**Pull request by bpedersen2**
Export author, co-authors and boardnumber as well.

### CODE REVIEW

File: CHANGES.rst [hunk 1]
```diff
@@ -19,6 +24,7 @@ Bugfixes
 - Fix error in HTTP API when Basic auth headers are present (:issue:`4123`,
   thanks :user:`uxmaster`)
 
+
```
Comment (ThiefMaster) [inline CHANGES.rst]: we usually just have one blank line between those versions, so please don't add another one here ;)

File: indico/modules/events/contributions/util.py [hunk 2]
```diff
@@ -107,7 +109,12 @@ def generate_spreadsheet_from_contributions(contributions):
                         'Session': c.session.title if c.session else None,
                         'Track': c.track.title if c.track else None,
                         'Materials': None,
-                        'Presenters': ', '.join(speaker.full_name for speaker in c.speakers)}
+                        'Presenters': ', '.join(speaker.full_name for speaker in c.speakers),
+                        'Authors': ', '.join(author.full_name for author in c.submitters
+                                             if author.author_type == AuthorType.primary),
+                        'Co-Authors': ', '.join(author.full_name for author in c.submitters
+                                                if author.author_type != AuthorType.primary),
+                        'Boardnumber': c.board_number}
```
Comment (ThiefMaster) [inline indico/modules/events/contributions/util.py]: I think spaces in CSV column headers are OK, so I'd rather go for `'Board number'` here.

"Boardnumber" sounds a bit "denglisch" ;)

File: indico/modules/events/contributions/util.py [hunk 3]
```diff
@@ -97,7 +98,8 @@ def generate_spreadsheet_from_contributions(contributions):
     """Return a tuple consisting of spreadsheet columns and respective
     contribution values"""
 
-    headers = ['Id', 'Title', 'Description', 'Date', 'Duration', 'Type', 'Session', 'Track', 'Presenters', 'Materials']
+    headers = ['Id', 'Title', 'Description', 'Date', 'Duration', 'Type', 'Session', 'Track', 'Presenters', 'Materials',
+               'Authors', 'Co-Authors', 'Boardnumber']
```
Comment (ThiefMaster) [inline indico/modules/events/contributions/util.py]: Should we include the board number at all if there aren't any contributions with one?

Would be pretty easy to check for it..

```python
has_board_numbers = any(c.board_number for c in contributions)
```

File: CHANGES.rst [hunk 4]
```diff
@@ -7,6 +7,12 @@ Version 2.2.5
 
 *Unreleased*
 
+Improvements
+^^^^^^^^^^^^
+
+- Add additional fields to the contribution csv/xlsx export (Authors,
+  Boardnumber) (:issue:`4148`, thanks :user:`bpedersen2`)
```
Comment (ThiefMaster) [inline CHANGES.rst]: "(authors, board number)"

File: CHANGES.rst (hunk 1 above)
Comment (bpedersen2) [reply CHANGES.rst]: fixed

File: indico/modules/events/contributions/util.py (hunk 3 above)
Comment (bpedersen2) [reply indico/modules/events/contributions/util.py]: Sounds like a good idea. Done

File: CHANGES.rst (hunk 4 above)
Comment (bpedersen2) [reply CHANGES.rst]: fixed

**Pull request merged**